genesyscloud/routing_queue_conditional_group_routing/resource_genesyscloud_routing_queue_conditional_group_routing_schema.go

```diff
@@ -0,0 +1,121 @@
+		},
+		SchemaVersion: 1,
+		Schema: map[string]*schema.Schema{
+			"queue_id": {
+				Description: "Id of the routing queue to which the rules belong",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"rules": {
+				Description: "The Conditional Group Routing settings for the queue.",
+				Type:        schema.TypeList,
+				Required:    true,
+				MinItems:    1,
+				MaxItems:    5,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"evaluated_queue_id": {
+							Description: "The queue being evaluated for this rule. For rule 1, this is always the current queue, so should not be specified.",
+							Type:        schema.TypeString,
+							Optional:    true,
+						},
+						"operator": {
+							Description:  "The operator that compares the actual value against the condition value. Valid values: GreaterThan, GreaterThanOrEqualTo, LessThan, LessThanOrEqualTo.",
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringInSlice([]string{"GreaterThan", "LessThan", "GreaterThanOrEqualTo", "LessThanOrEqualTo"}, false),
+						},
+						"metric": {
+							Description:  "The queue metric being evaluated. Valid values: EstimatedWaitTime, ServiceLevel.",
+							Type:         schema.TypeString,
+							Optional:     true,
+							Default:      "EstimatedWaitTime",
+							ValidateFunc: validation.StringInSlice([]string{"EstimatedWaitTime", "ServiceLevel"}, false),
+						},
+						"condition_value": {
+							Description:  "The limit value, beyond which a rule evaluates as true.",
+							Type:         schema.TypeFloat,
+							Required:     true,
+							ValidateFunc: validation.FloatBetween(0, 259200),
+						},
+						"wait_seconds": {
+							Description:  "The number of seconds to wait in this rule, if it evaluates as true, before evaluating the next rule. For the final rule, this is ignored, so need not be specified.",
+							Type:         schema.TypeInt,
+							Optional:     true,
+							Default:      2,
+							ValidateFunc: validation.IntBetween(0, 259200),
+						},
+						"groups": {
+							Type:        schema.TypeList,
+							Required:    true,
+							MinItems:    1,
+							Description: "The group(s) to activate if the rule evaluates as true.",
+							Elem:        memberGroupResource,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func RoutingQueueConditionalGroupRoutingExporter() *resourceExporter.ResourceExporter {
+	return &resourceExporter.ResourceExporter{
+		GetResourcesFunc: provider.GetAllWithPooledClient(getAllAuthRoutingQueueConditionalGroup),
+		RefAttrs: map[string]*resourceExporter.RefAttrSettings{
+			"queue_id":                 {RefType: "genesyscloud_routing_queue"},
+			"rules.evaluated_queue_id": {RefType: "genesyscloud_routing_queue"},
+		},
+		CustomAttributeResolver: map[string]*resourceExporter.RefAttrCustomResolver{
+			"rules.groups.member_group_id": {ResolverFunc: resourceExporter.MemberGroupsResolver},
```

Comment:
Should this be included in the RefAttrs map? Also, if this can reference a skill group, regular group, or team, you might need to use the resolver func `MemberGroupsResolver`

Reply:
I don't think I need to add this to the RefAttrs map, since the member_group can be a team, group or skill group, the custom resolver handles creating the reference. Also, is it not already using the resolver func `MemberGroupsResolver`

Reply:
Ah jeepers. My bad. I didn't read the code properly at all. Just saw that it wasn't in the refattrs map and thought it must have to be added. Nevermind! 😅